include/zephyr/sys/cbprintf_internal.h

```diff
@@ -382,6 +382,11 @@ do { \
 		 _name##_buf32)))
 #endif
 
+static inline bool ___is_null(void *p)
```

Comment:
Can you add a comment explaining why this is needed?

Reply:
done

Reply:
I think it would be better to explain it in details instead of linking to a GitHub issue, as all the information would be in the code.

Also, if you are changing this again, could you do `ALWAYS_INLINE` instead of simply `inline`? This will restore old behavior during debugging with no optimizations.

Reply:
Do we really want to put all explanation in the code in this Git era?  You can just `git blame` to see the commit message.  Nobody put comments with its associated commit messages on every line we change.

Reply:
It doesn't have to be as detailed as the commit message. But a summary of that would be great. Besides, if this ever gets modified, `git blame` would not point to this commit.

Reply:
Done.